File: foodify_mobile/lib/features/home/data/models/food_model.dart
import '../../domain/entities/food_entity.dart';

class FoodModel extends FoodEntity {
  const FoodModel({
    required super.id,
    required super.restaurantId,
    required super.name,
    required super.image,
    required super.description,
    required super.price,
    required super.category,
    required super.rating,
    required super.isAvailable,
    required super.isPopular,
    super.restaurantName,
    super.restaurantImage,
  });

  factory FoodModel.fromJson(Map<String, dynamic> json) {
    // Handle populated restaurantId
    String restaurantId = '';
    String? restaurantName;
    String? restaurantImage;

    if (json['restaurantId'] is String) {
      restaurantId = json['restaurantId'];
    } else if (json['restaurantId'] is Map) {
      restaurantId = json['restaurantId']['_id'] ?? '';
      restaurantName = json['restaurantId']['name'];
      restaurantImage = json['restaurantId']['image'];
    }

    return FoodModel(
      id: json['_id'] ?? '',
      restaurantId: restaurantId,
      name: json['name'] ?? '',
      image: json['image'] ?? '',
      description: json['description'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      category: json['category'] ?? '',
      rating: (json['rating'] ?? 0).toDouble(),
      isAvailable: json['isAvailable'] ?? true,
      isPopular: json['isPopular'] ?? false,
      restaurantName: restaurantName,
      restaurantImage: restaurantImage,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'restaurantId': restaurantId,
      'name': name,
      'image': image,
      'description': description,
      'price': price,
      'category': category,
      'rating': rating,
      'isAvailable': isAvailable,
      'isPopular': isPopular,
    };
  }
}

File: foodify_api/src/models/Food.model.ts
import mongoose from "mongoose";

const foodSchema = new mongoose.Schema({
    restaurantId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Restaurant",
        required: true
    },
    name: { type: String, required: true },
    image: { type: String, required: true },
    description: { type: String, required: true },
    price: { type: Number, required: true },
    category: { type: String, required: true },
    rating: { type: Number, default: 4.5 },
    isAvailable: { type: Boolean, default: true },
    isPopular: { type: Boolean, default: false },
}, { timestamps: true });

export const FoodModel = mongoose.model("Food", foodSchema);